src/steps/format-json.js

```diff
@@ -0,0 +1,53 @@
+import dasherize from "dasherize";
+import camelize from "../util/camelize";
+import Resource from "../types/Resource";
+import Collection from "../types/Collection";
+
+export function dasherizeResource(resource, registry) {
+  let exceptions = registry.behaviors(resource.type).dasherizeOutput.exceptions;
+  return transformResourceKeys(resource, dasherize, exceptions);
+}
+
+export function dasherizeResourceOrCollection(toDasherize, registry) {
+  return transformResourceOrCollection(toDasherize, dasherizeResource, registry);
+}
+
+export function camelizeResource(resource, registry) {
+  let exceptions = registry.behaviors(resource.type).dasherizeOutput._inverseExceptions;
+  return transformResourceKeys(resource, camelize, exceptions);
+}
+
+export function camelizeResourceOrCollection(toCamelize, registry) {
+  return transformResourceOrCollection(toCamelize, camelizeResource, registry);
+}
+
+function transformResourceKeys(resource, transformFn, exceptions) {
```

Comment:
Hm. I'm unclear on the spec here—can a relationship object’s `data` object contain `attributes`? The examples in the spec only ever show it with `type` and `id`. If there are no `attributes`, there’s nothing to do here.